### PR TITLE
Scenario comparison for deterministic models

### PR DESCRIPTION
**Pull request by pratikunterwegs**
This PR is WIP #221 and adds basic intervention scenario comparison functionality. The function `outcomes_averted()` offers options for calculating the differences in outcomes averted (cases, and deaths where applicable) between a baseline scenario (with parameter uncertainty), and multiple comparator scenarios (with identical parameter uncertainty).

- See the vignette on scenario modelling, `modelling_scenarios.Rmd` for how to use this.
- There's a limitation that this only works for models with parameter uncertainty. This is because the output format of a model with a single parameter set does not preserve model arguments/infection parameters which are used for matching. This in turn is because it was felt that users should be able to run a non-vectorised model and get simplified output that can be passed directly to plotting (see #176).
- There's no support for comparing scenarios where the axis of variation is an infection parameter, e.g. $R_0$ is one of 1.3, 2.0, 2.5 etc.
- Support for the Ebola model is forthcoming in a separate PR based on feedback from this one.

Happy to take feedback on the implementation here, and more general discussion on scenario comparison in #221.

### CODE REVIEW

File: R/scenario_comparison.R [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+#' # Defaults to summarise = TRUE
+#' outcomes_averted(
+#'   baseline = baseline,
+#'   scenarios = scenarios
+#' )
+#'
+#' # Set summarise = FALSE to apply custom summary functions
+#' outcomes_averted(
+#'   baseline = baseline,
+#'   scenarios = scenarios
```
Comment (joshwlambert) [inline R/scenario_comparison.R:128]: Should `summarise = FALSE` here?

File: vignettes/modelling_scenarios.Rmd [hunk 2]
```diff
@@ -607,12 +399,12 @@ The intervention scenarios modelled above suggest that:
...
 In this simple example we show how this need not always be the case, and that counter-intuitively, implementing more interventions can lead to a larger epidemic final size than implementing fewer interventions.
 
-This phenomenon depends on the baseline transmission rate of the infection, so we select a relatively low $R$ of 1.30 (corresponding to pandemic influenza), and a higher $R$ of `r round(r_estimate_mean, 2)` from the $R$ estimation step at the start of the vignette.
+This phenomenon depends on the baseline transmission rate of the infection, so we select a relatively low $R$ of 1.30 (corresponding to pandemic influenza), and a higher $R$ of 1.6 to illustrate the point.
 
 ```{r}
 # run each scenario for two values of R
 # no parameter uncertainty
-r_values <- c(1.3, round(r_estimate_mean, 2))
+r_values <- c(1.3, 1.6)
```
Comment (joshwlambert) [inline vignettes/modelling_scenarios.Rmd:407]: Leaving a comment here for figure 2 below. The x-axis labels are overlapping on the pkgdown site, making it hard to match the column with the intervention.

**Review by joshwlambert — APPROVED** [review]
@pratikunterwegs nice work. All looks good to me. The new documentation in the vignettes makes the new functionality very clear. Left a couple of minor comments in the code.

File: vignettes/modelling_scenarios.Rmd (hunk 2 above)
Comment (pratikunterwegs) [reply vignettes/modelling_scenarios.Rmd:407]: Yep - unsure how to fix this. I tried setting `n.dodge = 3` but this still has overlaps.

**Comment by pratikunterwegs** [comment]
Thanks @joshwlambert. I haven't found a good fix for the second issue, but I'm merging this now so I can move ahead with more scenario functionality.

**Pull request merged**